cmd/node_builder.go

```diff
@@ -68,10 +68,18 @@ type NodeBuilder interface {
 	// `Done`.
 	Component(name string, f func(builder NodeBuilder, node *NodeConfig) (module.ReadyDoneAware, error)) NodeBuilder
 
-	// MustNot asserts that the given error must not occur.
-	// If the error is nil, returns a nil log event (which acts as a no-op).
-	// If the error is not nil, returns a fatal log event containing the error.
-	MustNot(err error) *zerolog.Event
+	// Errors returns a channel that receives any unrecoverable errors encountered by the builder
+	Errors() <-chan error
```

Comment:
https://www.geeksforgeeks.org/embedding-interfaces-in-golang/

^ You can just embed the `ErrorAware` interface here :)